### DISCUSSION TITLE
How to add built-in shaders in an Android build

### DISCUSSION THREAD

**Discussion by dibenedetto** [atteneder/glTFast#688] [Q&A]
---------------------
EDIT:
I found that putting them into the "Alwais Included Shaders" solves the problem.
Anyway, is this the right solution?
---------------------

I recently found that a possible bug in using built-in shaders, in `BuiltInMaterialGenerator.cs` :

```
s_DefaultMaterial = GetPbrMetallicRoughnessMaterial();
s_DefaultMaterial.name = DefaultMaterialName;
```

If line 1 returns null we have a problem in line 2.

In my case, the app running on an Android device doesn't find the `glTF/PbrMetallicRoughness` shader, even if it is apparently included in glTFast package. If I try to copy the package Shader directory into project Assets, I get a conflict warning from the editor.

How will I ensure that those shaders are included in the Android build?

**Comment by atteneder** [comment]
Glad you found a solution, but it's not an ideal one, because it'll include many shader variants, even those you'll never need.

The [documentation about it](https://docs.unity3d.com/Packages/com.unity.cloud.gltfast@6.6/manual/ProjectSetup.html#materials-and-shader-variants) explains it in detail.
